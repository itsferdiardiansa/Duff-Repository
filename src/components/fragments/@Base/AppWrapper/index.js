/* eslint-disable no-unused-vars */
import {
  defineComponent,
  h,
  ref,
  getCurrentInstance,
  unref,
  computed,
} from 'vue'
import { useStore } from 'vuex'
import { useRouter } from 'vue-router'
import globalSetting from '@plugin/globalSetting'
import ErrorComponent from '@base/Error'
import subscribeFormAction from './subscribeFormAction'
import asyncModuleLoad from './asyncModuleLoad'

const AppWrapper = defineComponent({
  beforeCreate() {
    const {
      appContext: { app },
    } = getCurrentInstance()
    const store = useStore()
    const router = useRouter()

    store.subscribe(({ payload }, state) => {
      subscribeFormAction(payload, router)
    })

    // Handle async module load
    asyncModuleLoad()

    // register global mixin
    app.mixin(globalSetting)
  },
  beforeRouteEnter() {
    console.log('before router enter wrapper')
  },
  errorCaptured(error, component) {},
  setup(props, { slots }) {
    const store = useStore()

    const errorChunkModule = computed(() => store.getters['app/getChunks'])

    return () => {
      return unref(errorChunkModule).onError
        ? h(ErrorComponent, {})
        : slots.default()
    }
  },
})

export default AppWrapper
