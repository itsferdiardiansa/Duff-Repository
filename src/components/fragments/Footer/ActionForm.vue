<template>
  <Form ref="formEl" :model="state.form" @submit="handleSubmit">
    <FormControl
      label="Name"
      :rules="{
        name: [{ required: true, message: 'Footer name is required' }],
      }"
    >
      <Input placeholder="Footer name" v-model="state.form.name" autofocus />
    </FormControl>

    <FormControl
      label="Position"
      :colspan="3"
      :rules="{
        position: [{ required: true, message: 'Position is required' }],
      }"
    >
      <Select
        placeholder="Please select position"
        keyname="value"
        v-model="state.form.position"
        :items="[
          { value: '0', label: 'Position Top' },
          { value: '1', label: 'Position Right' },
          { value: '2', label: 'Position Bottom' },
          { value: '3', label: 'Position Left' },
        ]"
      />
    </FormControl>

    <FormControl :offset="3">
      <Button
        type="submit"
        :label="isCreate ? 'Create' : 'Update'"
        :variant="isCreate ? 'dark' : 'warning'"
        :icon="['fa', 'save']"
        :bold="true"
        :isLoading="isFetching"
      />
    </FormControl>
  </Form>
</template>
<script>
/* eslint-disable */
import { onMounted, reactive, ref, unref } from 'vue';
import Form, { FormControl, Input, Select } from '@common/Form';
import Button from '@common/Button';

export default {
  name: 'PartnerActionForm',
  components: {
    Button,
    Form,
    FormControl,
    Input,
    Select,
  },
  emits: ['submit'],
  props: {
    data: {
      type: Object,
      default: () => {},
    },
    isCreate: {
      type: Boolean,
      default: true,
    },
    isFetching: {
      type: Boolean,
      default: false,
    },
    withValidation: {
      type: Boolean,
      default: true,
    },
  },
  setup(props, { emit }) {
    const formEl = ref();
    const state = reactive({
      form: {
        name: '',
        position: '',
      },
    });

    const handleSubmit = () => {
      const { withValidation } = props;
      const form = unref(formEl);

      form.validate(valid => {
        if (valid || !withValidation) {
          emit('submit', state.form);
        }
      });
    };

    onMounted(() => {
      const { data, isCreate } = props;

      if (data && !isCreate) state.form = data;
    });

    return {
      formEl,
      state,
      handleSubmit,
    };
  },
};
</script>
