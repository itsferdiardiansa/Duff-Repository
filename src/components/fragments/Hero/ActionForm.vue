<template>
  <Form ref="formEl" :model="state.form" @submit="handleSubmit">
    <FormControl
      label="Title"
      :rules="{
        title: [{ required: true, message: 'Title is required' }],
      }"
    >
      <Input
        placeholder="Title"
        name="Title"
        v-model="state.form.title"
        autofocus
      />
    </FormControl>

    <FormControl
      label="Button URL"
      :rules="{
        button_url: [
          { required: true, message: 'Button URL is required' },
          { pattern: urlPattern, message: 'Please check the URL format' },
        ],
      }"
    >
      <Input
        placeholder="Button URL"
        v-model="state.form.button_url"
        autofocus
      />
    </FormControl>

    <FormControl
      label="Description"
      :rules="{
        description: [{ required: true, message: 'Description is required' }],
      }"
    >
      <Textarea name="description" v-model="state.form.description" />
    </FormControl>
    <FormControl
      label="Banner"
      :rules="{
        banner: [{ required: true, message: 'Banner is required' }],
      }"
    >
      <FileUpload name="banner" v-model="state.form.banner" />
    </FormControl>

    <FormControl
      label="Banner Mobile"
      :rules="{
        banner_mobile: [
          { required: true, message: 'Banner Mobile is required' },
        ],
      }"
    >
      <FileUpload name="banner_mobile" v-model="state.form.banner_mobile" />
    </FormControl>

    <FormControl :offset="3">
      <Button
        type="submit"
        variant="orange"
        :label="isCreate ? 'Create' : 'Update'"
        :icon="['fa', 'save']"
        :bold="true"
        :isLoading="isFetching"
      />
    </FormControl>
  </Form>
</template>
<script>
/* eslint-disable */
import { onMounted, reactive, ref, unref } from 'vue';
import Form, { FormControl, FileUpload, Input, Textarea } from '@common/Form';
import Button from '@common/Button';
import { urlPattern } from '@util/pattern';

export default {
  name: 'HeroActionForm',
  components: {
    Button,
    Form,
    FormControl,
    FileUpload,
    Input,
    Textarea,
  },
  emits: ['submit'],
  props: {
    data: {
      type: Object,
      default: () => {},
    },
    isCreate: {
      type: Boolean,
      default: true,
    },
    isFetching: {
      type: Boolean,
      default: false,
    },
    withValidation: {
      type: Boolean,
      default: true,
    },
  },
  setup(props, { emit }) {
    const formEl = ref();
    const state = reactive({
      form: {
        title: '',
        button_url: '',
        banner: '',
        banner_mobile: '',
        position: 0,
        is_active: 0,
        banner_type: 'image',
      },
    });

    const handleSubmit = () => {
      const { withValidation } = props;
      const form = unref(formEl);

      form.validate(valid => {
        if (valid || !withValidation) {
          emit('submit', state.form);
        }
      });
    };

    onMounted(() => {
      const { data, isCreate } = props;

      if (data && !isCreate) state.form = data;
    });

    return {
      formEl,
      state,
      handleSubmit,
      urlPattern,
    };
  },
};
</script>
