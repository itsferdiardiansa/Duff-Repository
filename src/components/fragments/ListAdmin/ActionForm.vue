<template>
  <pre>
    {{ JSON.stringify(state.form, null, 2) }}
  </pre>
  <Form ref="formEl" :model="state.form" @submit="handleSubmit">
    <FormControl
      label="Name"
      :rules="{
        name: [{ required: true, message: 'Name is required' }],
      }"
    >
      <Input placeholder="Role Name" v-model="state.form.name" autofocus />
    </FormControl>

    <FormControl
      label="Email"
      :rules="{
        email: [
          { required: true, message: 'Email is required' },
          { pattern: emailPattern, message: 'Please check the email format' },
        ],
      }"
    >
      <Input placeholder="Email" v-model="state.form.email" autofocus />
    </FormControl>

    <FormControl
      label="Role"
      :rules="{
        role_hash_id: [{ required: true, message: 'Role is required' }],
      }"
    >
      <div class="flex items-center">
        <Button
          :label="state.form.role ? 'Change role' : 'Choose role'"
          variant="dark"
          tabindex="-1"
          type="button"
          :icon="['fa', 'user-tag']"
          @click="handleSelectRole"
        />
      </div>
    </FormControl>

    <FormControl :offset="3">
      <Button
        type="submit"
        variant="orange"
        :label="isCreate ? 'Create' : 'Update'"
        :icon="['fa', 'save']"
        :bold="true"
        :isLoading="isFetching"
      />
    </FormControl>
  </Form>

  <Modal name="role-list">
    <div class="list">
      <template v-if="roleRequestStatus.fetch">
        <template v-for="item in range(8)" :key="item">
          <RectSkeleton width="100%" height="15px" />
        </template>
      </template>
      <template v-else-if="roleRequestStatus.error.status">
        <div class="error">
          <div class="wrapper">
            <div class="img">
              <img
                class="img-error"
                svg-inline
                src="@icon/error-chunk-load.svg"
              />
            </div>

            <div class="description">
              <span> Oops... something went wrong! </span>

              <Button
                variant="dark"
                :pill="true"
                :icon="['fa', 'redo-alt']"
                @click="fetchRole"
              />
            </div>
          </div>
        </div>
      </template>
      <template v-else>
        <template v-for="item in roles" :key="item.hash_id">
          <div class="item">
            <input
              type="radio"
              :id="item.hash_id"
              :value="item.hash_id"
              v-model="state.form.role_hash_id"
            />
            <label :for="item.hash_id" class="ml-2 cursor-pointer">{{
              item.name
            }}</label>
          </div>
        </template>
      </template>
    </div>

    <Pagination
      v-bind="rolePagination"
      @changePage="handlePageChange"
      v-if="!roleRequestStatus.fetch && !roleRequestStatus.error.status"
    />
  </Modal>
</template>
<script>
/* eslint-disable */
import { computed, nextTick, onMounted, reactive, ref, unref } from 'vue';
import { useStore } from 'vuex';
import Form, {
  FormControl,
  FileUpload,
  Input,
  Textarea,
  Select,
} from '@common/Form';
import Button from '@common/Button';
import { Rect as RectSkeleton } from '@common/Skeleton';
import { emailPattern } from '@util/pattern';
import Pagination from '@common/Pagination';
import http from '@service/http';

export default {
  name: 'AdminActionForm',
  components: {
    Button,
    Form,
    FormControl,
    FileUpload,
    Input,
    Textarea,
    Select,
    RectSkeleton,
    Pagination,
  },
  emits: ['submit'],
  props: {
    data: {
      type: Object,
      default: () => {},
    },
    isCreate: {
      type: Boolean,
      default: true,
    },
    isFetching: {
      type: Boolean,
      default: false,
    },
    withValidation: {
      type: Boolean,
      default: true,
    },
  },
  setup(props, { emit }) {
    const formEl = ref();
    const store = useStore();
    const pagination = reactive({ page: 1, limit: 10 });
    const state = reactive({
      form: {
        name: '',
        email: '',
        role_hash_id: null,
      },
    });

    const handleSubmit = () => {
      const { withValidation } = props;
      const form = unref(formEl);

      form.validate(valid => {
        if (valid || !withValidation) {
          emit('submit', state.form);
        }
      });
    };

    const roles = computed(() => {
      return store.getters['role/getItems'];
    });

    const rolePagination = computed(() => {
      return store.getters['role/getPagination'];
    });

    const roleRequestStatus = computed(() => {
      return store.getters['role/getRequestStatus'];
    });

    const fetchRole = () => {
      store.dispatch('role/fetchData', pagination);
    };

    const handleSelectRole = () => {
      fetchRole();

      SSModal.show('role-list', {
        title: 'Role List',
        footer: false,
        afterClose: () => {
          http.requestCancel.cancel('fetch role has been rejected');
        },
      });
    };

    const handlePageChange = params => {
      Object.assign(pagination, params);

      fetchRole();
    };

    nextTick(() => {
      const { data, isCreate } = props;

      if (data && !isCreate) {
        state.form = data;
        state.form.role_hash_id = data.role?.hash_id;
      }
    });

    return {
      formEl,
      state,
      roles,
      rolePagination,
      roleRequestStatus,
      fetchRole,
      handlePageChange,
      handleSubmit,
      emailPattern,
      handleSelectRole,
    };
  },
};
</script>
<style lang="scss" scoped>
.list {
  @apply grid grid-cols-2 gap-3;

  .item {
    @apply flex items-center;
  }

  .error {
    @apply col-span-2;

    .wrapper {
      @apply flex flex-col items-center;

      .img {
        svg {
          @apply w-60 h-60;
        }
      }

      .description {
        @apply flex items-center;

        span {
          @apply mr-2;
        }
      }
    }
  }
}
</style>
