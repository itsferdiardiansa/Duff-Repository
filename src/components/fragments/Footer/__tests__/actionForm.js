import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import AppWrapper from '@base/AppWrapper';
import ActionForm from '@fragment/Footer/ActionForm';
import footerModule from '@store/modules/footer';

describe('fragment/Footer/ActionForm', () => {
  let store;

  beforeAll(() => {
    store = createStore({
      modules: {
        footer: footerModule,
      },
    });
  });

  it('will render and match snapshot', () => {
    const Component = mount(<ActionForm />, {
      global: {
        plugins: [store],
      },
    });

    expect(Component.element).toMatchSnapshot();
  });

  it('will render with error validation on all fields and match snapshot', async () => {
    const Component = mount(<ActionForm />, {
      global: {
        plugins: [store],
      },
    });

    await Component.find('form').trigger('submit.prevent');

    // Total fields is 2 and shl=ould be 2
    const errorFields = Component.findAll(
      `.${Component.rootVM.prefixClass}-control--info`
    );

    expect(errorFields.length).toBe(2);

    await Component.vm.$nextTick(() => {
      expect(Component.element).toMatchSnapshot();
    });
  });

  it('will post data on success and bring up an alert then redirect it to the list', async () => {
    const Component = mount(
      <AppWrapper>
        <ActionForm />
      </AppWrapper>,
      {
        global: {
          plugins: [store],
        },
      }
    );

    const data = {};

    await Component.vm.$store.dispatch('footer/postData', {
      data,
      action: 'form.create',
      redirectUrl: '/footer',
    });
    expect(document.body).toMatchSnapshot();
    expect(mockRouterPush).toHaveBeenCalledWith('/footer');
  });

  it('will post data on failed and bring up an alert', async () => {
    const Component = mount(
      <AppWrapper>
        <ActionForm />
      </AppWrapper>,
      {
        global: {
          plugins: [store],
        },
      }
    );

    mockCreateFooter.mockReset();
    mockCreateFooter.mockRejectedValue({
      message: 'internal.error',
      result: {},
    });

    await Component.vm.$store.dispatch('footer/postData', {
      action: 'form.create',
    });

    expect(document.body).toMatchSnapshot();
  });

  it('will submit by pinning props as a callback', async () => {
    const handleSubmit = jest.fn();
    const formData = {
      tagline: 'tagline',
      site: 'https://google.com',
      description: '<p>Type your description here</p>',
      position: '1',
      ldp_id: '123456',
      is_active: '1',
      logo: '/9xbg6dsds==', // base64 format
      hero: '/7343bxx==', // base64 format
    };
    const Component = mount(
      <ActionForm model={formData} onSubmit={handleSubmit} />,
      { props: { withValidation: false }, global: { plugins: [store] } }
    );

    await Component.find('form').trigger('submit.prevent');

    expect(handleSubmit).toHaveBeenCalled();
  });

  it('will render the spinner during the request process', async () => {
    const Component = mount(<ActionForm />, {
      props: { isFetching: true },
      global: { plugins: [store] },
    });

    expect(
      Component.find(
        `[type="submit"] .${Component.rootVM.prefixClass}-button--spinner`
      ).exists()
    ).toBeTruthy();
    expect(Component.element).toMatchSnapshot();
  });

  it('will render the text and color for the form update', async () => {
    const Component = mount(<ActionForm />, {
      props: { isCreate: false },
      global: { plugins: [store] },
    });

    expect(Component.find(`[type="submit"]`).text()).toEqual('Update');
    expect(Component.find(`[type="submit"]`).attributes('class')).toContain(
      `${Component.rootVM.prefixClass}-button--orange`
    );
    expect(Component.element).toMatchSnapshot();
  });

  it('will overide local data when in update mode', async () => {
    const formData = {
      tagline: 'tagline',
      site: 'https://google.com',
      description: '<p>Type your description here</p>',
      position: '1',
      ldp_id: '123456',
      is_active: '1',
      logo: '/9xbg6dsds==', // base64 format
      hero: '/7343bxx==', // base64 format
    };
    const Component = mount(<ActionForm />, {
      props: { data: formData, isCreate: false },
      global: {
        plugins: [store],
      },
    });

    expect(Component.vm.state.form).toMatchObject(formData);
    expect(Component.vm.state).toMatchSnapshot();
  });

  it('will execute the form without validation', async () => {
    const handleSubmit = jest.fn();
    const Component = mount(<ActionForm onSubmit={handleSubmit} />, {
      props: { withValidation: false },
      global: {
        plugins: [store],
      },
    });

    await Component.find('form').trigger('submit.prevent');

    expect(handleSubmit).toHaveBeenCalled();
  });
});
