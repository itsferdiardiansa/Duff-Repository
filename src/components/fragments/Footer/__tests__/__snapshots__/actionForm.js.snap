// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Footer/ActionForm will overide local data when in update mode 1`] = `
Object {
  "form": Object {
    "description": <p>Type your description here</p>,
    "hero": "/7343bxx==",
    "is_active": "1",
    "ldp_id": "123456",
    "logo": "/9xbg6dsds==",
    "position": "1",
    "site": "https://google.com",
    "tagline": "tagline",
  },
}
`;

exports[`fragment/Footer/ActionForm will post data on failed and bring up an alert 1`] = `
<body>
  
  
  <div
    id="app"
  >
    
  
    <div
      <=""
      div=""
      id="plugin__12e5"
    >
      

    </div>
  </div>
  <div
    class="ss-notification-root"
    data-v-app=""
    role="alert"
  >
    
    <div
      class="ss-notification ss-notification--success slide-top-enter-from slide-top-enter-active"
    >
      <div
        class="ss-notification--icon"
      >
        <svg
          class=""
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="ss-notification--wrapper"
      >
        <!--v-if-->
        <!--v-if-->
        <div
          class="ss-notification--content"
        >
          <label />
        </div>
      </div>
    </div>
    
  </div>
</body>
`;

exports[`fragment/Footer/ActionForm will post data on success and bring up an alert then redirect it to the list 1`] = `
<body>
  
  
  <div
    id="app"
  >
    
  
    <div
      <=""
      div=""
      id="plugin__12e5"
    >
      

    </div>
  </div>
  <div
    class="ss-notification-root"
    data-v-app=""
    role="alert"
  >
    
    <div
      class="ss-notification ss-notification--success slide-top-enter-from slide-top-enter-active"
    >
      <div
        class="ss-notification--icon"
      >
        <svg
          class=""
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="ss-notification--wrapper"
      >
        <!--v-if-->
        <!--v-if-->
        <div
          class="ss-notification--content"
        >
          <label />
        </div>
      </div>
    </div>
    
  </div>
</body>
`;

exports[`fragment/Footer/ActionForm will render and match snapshot 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Name
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          placeholder="Footer name"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Position
    </div>
    <div
      class="ss-form-control--wrapper col-3"
    >
      <div
        class="ss-control--input"
      >
        <div
          class="ss-select"
        >
          <div
            class="ss-select--button"
            tabindex="0"
          >
            <span
              class="button-label"
            >
              Please select position
            </span>
            <span
              class="button-icon"
            >
              <svg
                class="h-4"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm231-113.9L103.5 277.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L256 226.9l101.6 101.6c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L273 142.1c-9.4-9.4-24.6-9.4-34 0z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="ss-select--dropdown slide-top-enter-from slide-top-enter-active"
            style="display: none;"
          >
            <ul
              class="dropdown-list"
            >
              
              <li
                class="dropdown-item"
              >
                Position Top
              </li>
              <li
                class="dropdown-item"
              >
                Position Right
              </li>
              <li
                class="dropdown-item"
              >
                Position Bottom
              </li>
              <li
                class="dropdown-item"
              >
                Position Left
              </li>
              <li
                class="dropdown-item"
              >
                Position Front
              </li>
              
            </ul>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--orange ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Create
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Footer/ActionForm will render the spinner during the request process 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Name
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          placeholder="Footer name"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Position
    </div>
    <div
      class="ss-form-control--wrapper col-3"
    >
      <div
        class="ss-control--input"
      >
        <div
          class="ss-select"
        >
          <div
            class="ss-select--button"
            tabindex="0"
          >
            <span
              class="button-label"
            >
              Please select position
            </span>
            <span
              class="button-icon"
            >
              <svg
                class="h-4"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm231-113.9L103.5 277.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L256 226.9l101.6 101.6c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L273 142.1c-9.4-9.4-24.6-9.4-34 0z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="ss-select--dropdown slide-top-enter-from slide-top-enter-active"
            style="display: none;"
          >
            <ul
              class="dropdown-list"
            >
              
              <li
                class="dropdown-item"
              >
                Position Top
              </li>
              <li
                class="dropdown-item"
              >
                Position Right
              </li>
              <li
                class="dropdown-item"
              >
                Position Bottom
              </li>
              <li
                class="dropdown-item"
              >
                Position Left
              </li>
              <li
                class="dropdown-item"
              >
                Position Front
              </li>
              
            </ul>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--orange ss-button--base text-bold"
        disabled=""
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <!--v-if-->
          
          
          <div
            class="ss-button--spinner"
          >
            <div
              class="ss-spinner"
              style="width: 20px; height: 20px;"
            >
              <div />
              <div />
              <div />
            </div>
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Footer/ActionForm will render the text and color for the form update 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Name
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          placeholder="Footer name"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Position
    </div>
    <div
      class="ss-form-control--wrapper col-3"
    >
      <div
        class="ss-control--input"
      >
        <div
          class="ss-select"
        >
          <div
            class="ss-select--button"
            tabindex="0"
          >
            <span
              class="button-label"
            >
              Please select position
            </span>
            <span
              class="button-icon"
            >
              <svg
                class="h-4"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm231-113.9L103.5 277.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L256 226.9l101.6 101.6c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L273 142.1c-9.4-9.4-24.6-9.4-34 0z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="ss-select--dropdown slide-top-enter-from slide-top-enter-active"
            style="display: none;"
          >
            <ul
              class="dropdown-list"
            >
              
              <li
                class="dropdown-item"
              >
                Position Top
              </li>
              <li
                class="dropdown-item"
              >
                Position Right
              </li>
              <li
                class="dropdown-item"
              >
                Position Bottom
              </li>
              <li
                class="dropdown-item"
              >
                Position Left
              </li>
              <li
                class="dropdown-item"
              >
                Position Front
              </li>
              
            </ul>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--orange ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Update
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Footer/ActionForm will render with error validation on all fields and match snapshot 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Name
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          placeholder="Footer name"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Footer name is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Position
    </div>
    <div
      class="ss-form-control--wrapper col-3"
    >
      <div
        class="ss-control--input"
      >
        <div
          class="ss-select"
        >
          <div
            class="ss-select--button"
            tabindex="0"
          >
            <span
              class="button-label"
            >
              Please select position
            </span>
            <span
              class="button-icon"
            >
              <svg
                class="h-4"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm231-113.9L103.5 277.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L256 226.9l101.6 101.6c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L273 142.1c-9.4-9.4-24.6-9.4-34 0z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="ss-select--dropdown slide-top-enter-from slide-top-enter-active"
            style="display: none;"
          >
            <ul
              class="dropdown-list"
            >
              
              <li
                class="dropdown-item"
              >
                Position Top
              </li>
              <li
                class="dropdown-item"
              >
                Position Right
              </li>
              <li
                class="dropdown-item"
              >
                Position Bottom
              </li>
              <li
                class="dropdown-item"
              >
                Position Left
              </li>
              <li
                class="dropdown-item"
              >
                Position Front
              </li>
              
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Position is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--orange ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Create
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;
